mathics/doc/common_doc.py

```diff
@@ -517,8 +509,151 @@ def __str__(self) -> str:
+            ]
+
+            def sorted_submodule(submodules):
+                return sorted(
+                    submodules,
+                    key=lambda submodule: submodule.sort_order
+                    if hasattr(submodule, "sort_order")
+                    else submodule.__name__,
+                )
+
+            # Add sections in the guide section...
+            for submodule in sorted_submodule(submodules):
+                if skip_module_doc(submodule, modules_seen):
+                    continue
+                elif IS_PYPY and submodule.__name__ == "builtins":
+                    # PyPy seems to add this module on its own,
+                    # but it is not something that can be importable
+                    continue
 
-def sorted_chapters(chapters: list) -> list:
+                submodule_name = get_doc_name_from_module(submodule)
+                if submodule_name in submodule_names_seen:
+                    continue
+                section = self.doc.add_section(
+                    self,
+                    submodule_name,
+                    submodule,
+                    operator=None,
+                    is_guide=False,
+                    in_guide=True,
+                )
+                modules_seen.add(submodule)
+                submodule_names_seen.add(submodule_name)
+                if guide_section is not None:
+                    guide_section.subsections.append(section)
+
+                builtins = builtins_by_module.get(submodule.__name__, [])
+                subsections = [builtin for builtin in builtins]
+                for instance in subsections:
+                    if hasattr(instance, "no_doc") and instance.no_doc:
+                        continue
+
+                    name = instance.get_name(short=True)
+                    if name in submodule_names_seen:
+                        continue
+
+                    submodule_names_seen.add(name)
+                    modules_seen.add(instance)
+
+                    self.doc.add_subsection(
+                        self,
+                        section,
+                        name,
+                        instance,
+                        instance.get_operator(),
+                        in_guide=True,
+                    )
+        else:
+            if not builtins:
+                return
+            sections = [
+                builtin for builtin in builtins if not skip_doc(builtin.__class__)
+            ]
+            self.doc.doc_sections(sections, modules_seen, self)
+        builtin_part.chapters.append(self)
+
+    def get_tests(self) -> Iterator[Tests]:
+        """
+        Generator which returns all of the Mathics3 Doctest-type tests for a given chapter.
+        It is assumed that everything has been "gathered" or parsed and
+        extracted from all modules previously.
+        """
+
+        for section in self.all_sections:
+            if section.installed:
+                if section.in_guide:
```

Comment:
I do not understand this logic: For some reason, a GuideSection is populated with subsections
that are already loaded in the Chapter as sections. Should not be easier to handle a GuideSection as a (shallow) section without subsections?

Reply:
I don't recall what's up here. It may be wrong.